exercises/react/example.cr

```diff
@@ -0,0 +1,76 @@
+class Cell
+  @dependents = [] of ComputeCell
+
+  def initialize(initial : Int32)
+    @value = initial
+  end
+
+  getter :value
+
+  protected getter :dependents
+end
+
+class InputCell < Cell
+  def value=(new_value : Int32)
+    @value = new_value
+    @dependents.each(&.propagate)
+    @dependents.each(&.fire_callbacks)
+  end
+end
+
+class ComputeCell < Cell
+  alias CallbackID = Int32
+
+  @callbacks_given = 0
+  @callbacks = {} of CallbackID => Int32 ->
+  @last_callback_value : Int32
+
+  def initialize(new_value : -> Int32, *inputs : Cell)
+    super(new_value.call)
+    @new_value = new_value
+    @last_callback_value = @value
+    inputs.each { |i| i.dependents << self }
+  end
+
+  def add_callback(&block : Int32 ->) : CallbackID
+    @callbacks_given += 1
+    @callbacks[@callbacks_given] = block
+    @callbacks_given
+  end
+
+  def remove_callback(id : CallbackID)
+    @callbacks.delete(id)
+  end
+
+  def propagate
```

Comment:
Actually, funnily enough, that's doesn't solve the problem. That prevents someone from e.g. creating a cell since they can't do `InputCell.new`, but if they have a `ComputeCell`, they can freely call `propagate` on it. OK, so I guess I'm still looking for a solution on this one (but I wouldn't hold up the PR on it if nobody offers one)